src/panels/lovelace/hui-entity-filter-card.js

```diff
@@ -1,55 +1,113 @@
-`;
-  }
-
   static get properties() {
     return {
-      hass: Object,
-      config: Object,
+      hass: {
+        type: Object,
+        observer: '_hassChanged'
+      },
+      config: {
+        type: Object,
+        observer: '_configChanged'
+      }
     };
   }
 
   getCardSize() {
-    // +1 for the header
-    return 1 + this._getEntities(this.hass, this.config.filter).length;
+    return this.lastChild ? this.lastChild.getCardSize() : 1;
   }
 
-  // Return a list of entities based on a filter.
-  _getEntities(hass, filter) {
-    const filters = [];
+  // Return a list of entities based on filters.
+  _getEntities(hass, filterList) {
+    const entities = new Set();
+    filterList.forEach((filter) => {
+      const filters = [];
+      if (filter.domain) {
+        filters.push(stateObj => computeStateDomain(stateObj) === filter.domain);
+      }
+      if (filter.entity_id) {
+        filters.push(stateObj => this._filterEntityId(stateObj, filter.entity_id));
+      }
+      if (filter.state) {
+        filters.push(stateObj => stateObj.state === filter.state);
+      }
 
-    if (filter.domain) {
-      const domain = filter.domain;
-      filters.push(stateObj => computeStateDomain(stateObj) === domain);
+      Object.values(hass.states).forEach((stateObj) => {
+        if (filters.every(filterFunc => filterFunc(stateObj))) {
+          entities.add(stateObj.entity_id);
+        }
+      });
+    });
+    return Array.from(entities);
+  }
+
+  _filterEntityId(stateObj, pattern) {
+    if (pattern.indexOf('*') === -1) {
+      return stateObj.entity_id === pattern;
     }
+    const regEx = new RegExp(`^${pattern.replace(/\*/g, '.*')}$`);
+    return stateObj.entity_id.search(regEx) === 0;
+  }
 
-    if (filter.state) {
-      const state = filter.state;
-      filters.push(stateObj => stateObj.state === state);
+  _configChanged(config) {
+    if (this.lastChild) {
+      this.removeChild(this.lastChild);
     }
+    let error = null;
+    let cardConfig;
+    let tag = config.card ? computeCardElement(config.card) : 'hui-entities-card';
 
-    return Object.values(hass.states)
-      .filter(stateObj => filters.every(filterFunc => filterFunc(stateObj)))
-      .map(stateObj => stateObj.entity_id);
+    if (tag === null) {
+      error = `Unknown card type encountered: "${config.card}".`;
+    } else if (!customElements.get(tag)) {
+      error = `Custom element doesn't exist: "${tag}".`;
+    } else if (!config.filter || !Array.isArray(config.filter)) {
+      error = 'No or incorrect filter.';
+    }
+    if (error) {
+      tag = 'hui-error-card';
+      cardConfig = { error };
+    } else {
+      cardConfig = this._computeCardConfig(config);
+    }
+
+    const element = document.createElement(tag);
+    element.config = cardConfig;
+    element.hass = this.hass;
+    this.appendChild(element);
   }
 
-  _computeCardConfig(hass, config) {
-    return Object.assign({}, config.card_config || {}, {
```

Comment:
You broke this. We use `card_config` as the basis for config for next card. We should not copy the filter config. I'll fix this.